examples/plot_pulse_buffers.py

```diff
@@ -0,0 +1,96 @@
+from typing import Iterable
+
+import numpy as np
+from matplotlib import pyplot as plt
+from matplotlib import ticker
+from qat.purr.backends.echo import EchoEngine, get_default_echo_hardware
+from qat.purr.backends.live import LiveDeviceEngine
+from qat.purr.compiler.emitter import InstructionEmitter
+from qat.purr.compiler.runtime import get_builder
+from qat.purr.integrations.qasm import Qasm3Parser
+
+np.set_printoptions(
+    edgeitems=3000000, linewidth=100000000, formatter=dict(float=lambda x: "%.3g" % x)
+)
+
+import logging
+
+logging.disable()
+
+
+def plot_physical_buffers(
```

Comment:
If you want to show this off build a throwaway engine and use the normal pipeline. While these are the seperate components to do it, the code itself is very much not what you should do and putting it as an example gives the wrong idea.

Reply:
yeah fair, I understand why to build this. The thinking was more behind breaking down the various buffer building steps. But think it's clear enough from here tbf, might add a comment to clarify what each step is